docs/subnets/customize-a-subnet.md

```diff
@@ -440,6 +441,93 @@ In addition to the AllowList interface, the FeeConfigManager adds the following
+
+These upgrades can be specified in a file named `upgrade.json` placed in the same directory as [`config.json`](#chain-configs) using the following format:
+
+```json
+{
+  "precompileUpgrades": [
+    {
+      "<precompileName>": {
+        "blockTimestamp": 1668950000, // unix timestamp precompile should activate at
+        "precompileOption": "value" // precompile specific configuration options, eg. "adminAddresses"
+      }
+    }
+  ]
+}
+```
+
+To disable a precompile, the following format should be used:
+
+```json
+{
+  "precompileUpgrades": [
+    {
+      "<precompileName>": {
+        "blockTimestamp": 1668950000, // unix timestamp the precompile should deactivate at
+        "disable": true
+      }
+    }
+  ]
+}
+```
+
+Each item in `precompileUpgrades` must specify exactly one precompile to enable or disable and the block timestamps must be in increasing order. Once an upgrade has been activated (a block after the specified timestamp has been accepted), it must always be present in `upgrade.json` exactly as it was configured at the time of activation (otherwise the node will refuse to start).
+
+Enabling and disabling a precompile is a network upgrade and should always be done with caution.
+
+:::danger
+
+For safety, you should always treat `precompileUpgrades` as append-only.
+
+As a last resort measure, it is possible to abort or reconfigure a precompile upgrade that has not been activated since the chain is still processing blocks using the prior rule set.
+
+:::
+
+If aborting an upgrade becomes necessary, you can remove the precompile upgrade from `upgrade.json` from the end of the list of upgrades. As long as the blockchain has not accepted a block with a timestamp past that upgrade's timestamp, it will abort the upgrade for that node.
+
+### Example
+
+```json
+{
+  "precompileUpgrades": [
+    {
+      "feeManagerConfig": {
+        "blockTimestamp": 1668950000,
+        "adminAddresses": ["0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"]
+      }
+    },
+    {
+      "txAllowListConfig": {
+        "blockTimestamp": 1668960000,
+        "adminAddresses": ["0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"]
+      }
+    },
+    {
+      "feeManagerConfig": {
+        "blockTimestamp": 1668970000,
+        "disable": true
+      }
+    }
+  ]
+}
+```
+
+This example enables the `feeManagerConfig` at the first block with timestamp >= `1668950000`, enables `txAllowListConfig` at the first block with timestamp >= `1668960000`, and disables `feeManagerConfig` at the first block with timestamp >= `1668970000`.
+
+When the precompile is disabled, its storage will be wiped so it is necessary to re-apply the desired configuration.
```

Comment:
```suggestion
When a disabled precompile has taken effect after its `blockTimestamp` has passed, its storage will be wiped. If you want to reenable it, you will need to treat it as a new configuration.
```
@darioush how do you think about this?

Reply:
How about:

When a precompile disable takes effect (ie., after its `blockTimestamp` has passed), its storage will be wiped. If you want to reenable it, you will need to treat it as a new configuration.

Reply:
Great